src/Diagnostics.RuntimeHost/Extensions/EncyptedConfigProviderExtensions.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Diagnostics.Logger;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class EncyptedConfigProviderExtensions
+    {
+        public static IConfigurationBuilder AddEncryptedProvider(this IConfigurationBuilder builder, string encryptionKey, string initializationVector, string file)
+        {
+            return builder.Add(new EncryptedConfigProvider(encryptionKey, initializationVector, file));
+        }
+    }
+
+    public class EncryptedConfigProvider : ConfigurationProvider, IConfigurationSource
+    {
+        static string EncryptionKey;
+
+        static string EncryptedFile;
+
+        static string InitializationVector;
+
+        public EncryptedConfigProvider()
+        {
+
+        }
+
+        public EncryptedConfigProvider(string configEncryptionKey, string configInitVector, string file)
+        {
+            EncryptionKey = configEncryptionKey;
+            EncryptedFile = file;
+            InitializationVector = configInitVector;
+        }
+
+        public override void Load()
+        {
+            Data = DecryptSettings();           
+        }
+
+        private IDictionary<string, string> DecryptSettings()
+        {
+            var config = new Dictionary<string, string>();
+            try
+            {
+                using (StreamReader reader = new StreamReader(EncryptedFile))
+                {
+                    string json = reader.ReadToEnd();
+                    var result = JsonConvert.DeserializeObject<Dictionary<string, string>>(json);
+
+                    foreach (KeyValuePair<string, string> item in result)
+                    {
+                        var plainText = DecryptSetting(item.Value);
```

Comment:
What happens if some weird failure comes while attempting to decrypt a bad value? Will we still add the other values to the configuration object?

Reply:
Ideally it should skip the current setting and proceed to other good values. I will add code to handle this.